### ISSUE TITLE
Can't discard/reset our initial login views, need some assistance with this

### ISSUE THREAD

**Issue by ajonno** [expo/ex-navigation#434]
hey folks, having some trouble figuring out how to reset our route(s) as part of a very standard login flow (sign up, login, -> home screen). 

ive tried to work out 'immediatelyResetStack' (https://github.com/exponent/ex-navigation/issues/30) but can't seem to get it to work.

we have a separate Routes.js file that makes all of the registered views available no probs. Navigating around the app is fine. it's just this item of removing / not allowing a user to go back (which they can do by swiping / hitting back button) to any of the login/signup views.

really appreciate help with this.

![image](https://cloud.githubusercontent.com/assets/3005269/24088760/96de0d46-0d82-11e7-9f00-957e2b70428d.png)

**Comment by amanthegreatone** [comment]
looking to implement something similar. With Android you could just finish the activity but haven't found a way to do this in RN till now with any of the navigation solutions.

**Comment by MajorJ** [comment]
Hi ajonno,

I went through this exact dilemma myself a few weeks back with my app. I found the best solution to be to take the log in screen out of the app's main navigation stack entirely.

I'm using Redux and storing the logged-in status in the app's state. My app's got a top-level component that basically says if not logged in -> render the sign in screen, and if logged in -> render the home screen of the app, which in my case contains a drawer nav. That way, it's impossible for the user to go back to the sign in screen, as it's not part of the navigation stack at all.

**Comment by ajonno** [comment]
thanks for this info appreciate it. unless im misunderstanding your description, your flow is only 2 views (a login view and then a welcome view for successful signin).

lets say that the user does need to login. so they land on the "gday" (first) page and they tap on the Login button which takes them to the Login page. i can't "pop"/remove (or can i ?) the first page, since the user might navigate back to that first page - to then go to the Sign Up page. per my diagram.

the thing i can't figure out is how to remove that first page from the stack.

**Comment by MajorJ** [comment]
You're right that you'll need a navigator to pop/push your 'gday', 'sign up' and 'sign in' routes - but you can still use the same basic idea. In your case though you'll need the 'gday' route to have its own, *separate* stack navigator to handle all the navigation that needs to happen before the user logs in.

So essentially you'd have a component whose render method would include something like:

`render() {
    this.props.signedIn ? <Home /> : <Gday />
}`

'Gday' would have a stack navigator to handle navigation between Gday, SignIn and SignUp, and Home would have a *different* stack navigator to handle navigation once the user logs in. So when a user logs in, the <Gday /> component gets unmounted and replaced by the <Home /> component, with its stack navigator containing the home screen as its first route.

The only thing you'll have to handle is the transition between the two components, as brief code above wouldn't give you the usual animation you get with navigation.

**Comment by ajonno** [comment]
Thanks this works nicely
